src/object.rs

```diff
@@ -1153,19 +1264,33 @@ pub trait SecretKeyFactory {
         }
     }
 
+    /// Helper to set a key on a key object, individual factories can override
+    /// this function to ensure the key is appropriately formed for the type.
+    /// For example by checking the buffer length.
     fn set_key(&self, obj: &mut Object, key: Vec<u8>) -> Result<()> {
         let keylen = key.len();
         obj.set_attr(Attribute::from_bytes(CKA_VALUE, key))?;
         self.set_key_len(obj, keylen)?;
         Ok(())
     }
 
+    /// Helper to allow other mechanisms select the correct key length.
```

Comment:
```suggestion
    /// Helper to allow other mechanisms to select the correct key length.
```

Reply:
There is no need to add a to, it's correct either way.